x/dex/keeper/integration_placelimitorder_test.go

```diff
@@ -243,6 +243,57 @@ func (s *DexTestSuite) TestPlaceLimitOrderInsufficientFunds() {
 	s.assertAliceLimitSellFails(err, "TokenA", 0, 10)
 }
 
+func (s *DexTestSuite) TestPlaceLimitOrderGTCWithDustMinAvgPriceFails() {
+	s.fundAliceBalances(1, 0)
+	s.fundBobBalances(0, 1)
+	// GIVEN LP liq at 148.37-148.42 (with dust)
+	s.bobDeposits(
+		NewDepositInt(sdkmath.ZeroInt(), sdkmath.NewInt(1), 50_000, 1),
+		NewDepositInt(sdkmath.ZeroInt(), sdkmath.NewInt(1), 50_001, 1),
+		NewDepositInt(sdkmath.ZeroInt(), sdkmath.NewInt(1), 50_002, 1),
+		NewDepositInt(sdkmath.ZeroInt(), sdkmath.NewInt(10), 50_003, 1),
+	)
+	// THEN alice GTC limitOrder minAvgPrice == limitPrice fails
+	limitPrice := math_utils.MustNewPrecDecFromStr("0.006737")
+	_, err := s.msgServer.PlaceLimitOrder(s.Ctx, &types.MsgPlaceLimitOrder{
+		Creator:             s.alice.String(),
+		Receiver:            s.alice.String(),
+		TokenIn:             "TokenA",
+		TokenOut:            "TokenB",
+		LimitSellPrice:      &limitPrice,
+		AmountIn:            sdkmath.NewInt(2000),
+		OrderType:           types.LimitOrderType_GOOD_TIL_CANCELLED,
+		MinAverageSellPrice: &limitPrice,
+	})
+	s.ErrorIs(err, types.ErrLimitPriceNotSatisfied)
```

Comment:
I think the error you are getting this place not because of the new logic, but because of the rounding nature. And thats why
1) LimitPrice converted from price to tick to tick and back in a msgHandler, and in the code limitPrice lower than MinAverageSellPrice (despite the fact that you set them the same)
`006736936405289611298728644` < `0.006737`
2) On every swap iteration we check 
```
		if limitPrice != nil && liq.Price().LT(*limitPrice) {
			break
		}
```
in other words we swap only and only if liq.Price>limitPrice

If every price > limitPrice, it's impossible the `avg deal true price` < limitPrice.
swap debug via 4 deposits:
```
price:  0.006738957688326027277211891
inAmount:  149 outAmount:  1
price:  0.006738283859940033273884502
inAmount:  149 outAmount:  1
price:  0.006737610098930140259858517
inAmount:  149 outAmount:  1
price:  0.006736936405289611298728644
inAmount:  1485 outAmount:  10
```
seems like only because of accumulated rounding errors we got realsellprice 13/1932=0.006728778467908902691511387 < limitPrice

I would suggest you to add two (at least) test cases, i mean and fail and success 
1) with bigger deposits and swaps amounts
2) with a large spread between ticks to make the meaning of `MinAverageSellPrice` more obvious

Reply:
This is the intended functionality. The goal was to be able to trade through a specific tick but tolerate a certain amount of price increase due to rounding. 
There is another possible use case, which I believe you are describing where `MinAverageSellPrice` < `LimitSellPrice`. I will add some tests for this.

Reply:
Oh, I see. I thought the "another possible use case" is the intended functionality :) Something like: "I want to sell my 10BTC for 500USDT and i do not care about highest and lowest price, i just want my average". Confused with the name. If you wanna just be tolerant to the rounding error, did you think about another approach, e.g. instead of 'MinAverageSellPrice', introduce the bool flag "beRoundingTolerant" and do not check at all truePrice if `true`, or may be `roundingTollerance` as offset ticks, i.e. the limit of rounding error which is allowed to bypass, in the nutshell the second approach is close to what you did, but it looks more intuitive for me.

Reply:
We went back and forth on this a lot. Just being ok with all rounding allows the possibility of front running, since there is no cap on how bad rounding can get. So it's better Passing an explicit value is much safer. Both use cases are valid. I will update docs to better explain the usage for this param